src/PointwiseFunctions/AnalyticSolutions/RelativisticEuler/TovStar.cpp

```diff
@@ -459,6 +459,39 @@ void TovVariables<DataType, Region>::operator()(
   }
 }
 
+template <typename DataType, StarRegion Region>
+void TovVariables<DataType, Region>::operator()(
+    const gsl::not_null<tnsr::i<DataType, 3>*> dcoords_pressure,
+    [[maybe_unused]] const gsl::not_null<Cache*> cache,
+    Tags::DcoordsPressure<DataType> /*meta*/) const {
+  const auto& dr_pressure =
+      get(cache->get_var(*this, Tags::DrPressure<DataType>{}));
+
+  // This function is not required for solving TOV equations themselves, but
+  // needed for computing magnetic fields in the MagnetizedTov initial data.
+  if constexpr (Region == StarRegion::Exterior) {
+    (*dcoords_pressure).get(0) = 0.0;
+    (*dcoords_pressure).get(1) = 0.0;
+    (*dcoords_pressure).get(2) = 0.0;
+  } else {
+    for (size_t i = 0; i < get_size(radius); ++i) {
+      const auto& ri = get_element(radius, i);
+      // dp/dx^i goes to zero at the center
+      if (UNLIKELY(ri < 1e-15)) {
```

Comment:
You don't need to do this branching, just add an `if constexpr (Region == StarRegion::Center)` clause where you set the pressure gradient at the center to zero. Then remove the loop over grid points.